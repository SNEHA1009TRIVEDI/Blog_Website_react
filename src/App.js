import './App.css';
// import Header from './Components/Header/Header';
// import Footer from './Components/Footer/Footer';
// import Body from './Body/Body';
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { BlogPostProvider } from "./contexts/BlogPostContext";

import { ERROR_ROUTE, HOME_ROUTE } from "./constants/routes.jsx";
import  Error  from "./pages/Error/index";
import Home  from "./pages/Home/index";
import  PageNotFound  from "./pages/PageNotFound/index";

function App() {
  return (
  //   <div className="App">
  //   <header>
  //     <Header />
  //   </header>
  //    <Body/>
  //   <footer>
  //     <Footer />
  //   </footer> 
  // </div>
  <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path={HOME_ROUTE} element={
              <BlogPostProvider>
                <Home />
              </BlogPostProvider>
            } />
          <Route path={`${ERROR_ROUTE}/:errorCode?`} element={<Error />} />
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
