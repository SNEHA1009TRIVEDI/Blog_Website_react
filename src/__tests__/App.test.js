import { fireEvent, render } from "@testing-library/react";
import React from "react";

import App from "../App.js";

describe("App", () => {
    it("should renders without crashing ", () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
    
});
