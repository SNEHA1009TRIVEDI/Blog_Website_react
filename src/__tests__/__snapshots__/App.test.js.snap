// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders without crashing  1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="App"
    >
      <header>
        <nav
          class="body-padding"
        >
          <div>
            <h1>
              The Artifact
            </h1>
            <p>
              <em>
                Culture & Art blog
              </em>
            </p>
          </div>
          <ul>
            <li>
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <div
        class="body-padding home-page"
      >
        <div
          class="loading-screen"
        >
          Loading!!!
        </div>
      </div>
      <footer>
        <div
          class="body-padding footer"
        >
          Â© Sneha Trivedi 2023
        </div>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
